### DISCUSSION TITLE
Suggestion: allow an Oracle for OTC-trading

### DISCUSSION THREAD

**Discussion by ubiubi18** [idena-network/idena-desktop#540] [Ideas]
Imagine a Plattform for decentralized OTC-trading for all major cryptocurrencies - based on idena-oracles

The seller sets up an oracle, where s/he assigns funding in $idna as security for the trade. The seller desposits $idna with higher value in $idna than the planed trade (for example eth/btc) itself has. The question the oracles is supposed to answer is: Witness specific seller and buyer adresses in block explorer. Did that seller address receive buyers payment (for example in btc) within x hours? Did that buyers adress receive sellers delivery (for example eth) in exchange within y hours? 

If the transactions did indeed happen, the oracle of randomly chosen participants aproves the trades - and the seller instandly gets back its assigned $idna. If the buyer didnt pay, the seller gets back $idna too. But in case the buyer did pay and the seller didnt deliver, the oracle votes to burn the sellers $idna-security as punishment - or to send it to the predefined idena-adress of the buyer as compensation.

And now lets do it with people providing liquidity and data for all those otc-markets - in exchange for rewards from the idena-zero-wallet...?

**Comment by midenaio** [comment]
Technically it's possible to swap iDNA<>BTC (or ETH or other assets). Oracles can confirm the BTC settlement. Once the voting is finished the iDNA will be unlocked/refunded. However such a swap will take several days which may be not very practical.
Converting it to discussions.
